File: solutions/738/main.py
from typing import List 

class Solution:
    def monotoneIncreasingDigits(self, n: int) -> int:
        strNum = list(str(n))
        for i in range(len(strNum)-1, 0, -1):
            if int(strNum[i-1]) > int(strNum[i]):
                strNum[i-1] = str(int(strNum[i-1]) - 1)
                strNum[i:] = '9' * (len(strNum) - i)
        return int("".join(strNum))

if __name__ == "__main__":
    n = 10
    s = Solution()
    assert s.monotoneIncreasingDigits(n) == 9

File: solutions/738/main.cpp
/* ************************************************************************
> File Name:     main.cpp
> Author:        Sean
> Project:    LeetCode
> Created Time:  Wed Nov 10 08:49:30 2021
> Description:   
 ************************************************************************/
#include <string> 
#include <gtest/gtest.h> 

using namespace std; 

class Solution {
public:
    int monotoneIncreasingDigits(int n) {
        string strNum = to_string(n);
        int flag = strNum.size();
        for (int i = strNum.size()-1; i > 0; i--) {
            if (strNum[i-1] > strNum[i]) {
                flag = i;
                strNum[i-1]--;
            }
        }
        for (int i = flag; i < strNum.size(); i++) {
            strNum[i] = '9';
        }
        return stoi(strNum);
    }
};

TEST(Test738, SimpleTest) {
    const int n = 10; 
    Solution s; 
    EXPECT_EQ(s.monotoneIncreasingDigits(n), 9);
}

int main() {
    testing::InitGoogleTest();
    return RUN_ALL_TESTS();
}
